robot_smach_states/src/robot_smach_states/count_objects_on_location.py

```diff
@@ -0,0 +1,93 @@
+
+        rospy.loginfo("Segmented %d objects" % len(object_classifications))
+        if object_classifications:
+            # object_classifications = self.robot.ed.classify(ids=segmented_object_ids)
+
+            for idx, obj in enumerate(object_classifications):
+                rospy.loginfo("   - Object {i} is a '{t}' (prob: {p}, ID: {id})".format(i=idx, t=obj.type, id=obj.id, p=obj.probability))
+
+            over_threshold = object_classifications
+            if self.threshold:
+                over_threshold = [obj for obj in object_classifications if obj.probability >= self.threshold]
+
+            dropped = {obj.id: obj.probability for obj in object_classifications if obj.probability < self.threshold}
+            rospy.loginfo("Dropping {l} entities due to low class. score (< {th}): {dropped}"
+                          .format(th=self.threshold, dropped=dropped, l=len(dropped)))
+
+            object_classifications = over_threshold
+
+            list_objects = [obj for obj in object_classifications if obj.type == self.object_type.resolve()]
+            num_objects = len(list_objects)
+            rospy.loginfo("Counted {} objects matching the query".format(num_objects))
+            self.num_objects_designator.write(num_objects)
+            return 'done'
+        else:
+            return 'failed'
+
+
+# Standalone testing -----------------------------------------------------------------
+
+class InspectAndCount(smach.StateMachine):
+    def __init__(self, robot, where_to_count_designator, type_to_count_designator, count_designator):
+        smach.StateMachine.__init__(self, outcomes=['Done', 'Aborted'])
+
+        entities = ds.VariableDesignator([], resolve_type=[ClassificationResult])
+
+        with self:
+            smach.StateMachine.add("INSPECT_TABLE", states.Inspect(robot=robot, entityDes=where_to_count_designator,
+                                                                   objectIDsDes=entities,
+                                                                   searchArea="on_top_of",
+                                                                   navigation_area="in_front_of"),
+                                   transitions={"done": "COUNT",
+                                                "failed": "Aborted"})
+
+            smach.StateMachine.add("COUNT",
+                                   CountObjectsOnLocation(robot,
+                                                          location=where_to_count_designator,
+                                                          segmented_objects_designator=entities,
+                                                          object_type=type_to_count_designator,
+                                                          num_objects_designator=count_designator.writeable),
+                                   transitions={'done': 'Done',
+                                                'failed': 'Aborted'})
+
+
+if __name__ == "__main__":
```

Comment:
As far as I can see it now this does not work out of the box. Furthermore such a test would need for example a specific location, which is not desired, since this location might not be present in each environment. For this reason this part is left out.